flexbox/src/main/java/com/google/android/flexbox/FlexboxLayoutManager.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.flexbox;
+
+import android.content.Context;
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.AttributeSet;
+
+/**
+ * LayoutManager for the {@link RecyclerView}. This class is intended to be used within a
+ * {@link RecyclerView} and offers the same capabilities of measure/layout its children
+ * as the {@link FlexboxLayout}.
+ */
+public class FlexboxLayoutManager extends RecyclerView.LayoutManager {
+
+    private RecyclerView mRecyclerView;
+
+    /**
+     * The current value of the {@link FlexDirection}, the default value is {@link
+     * FlexDirection#ROW}.
+     */
+    private int mFlexDirection;
+
+    /**
+     * The current value of the {@link FlexWrap}, the default value is {@link FlexWrap#WRAP}.
+     */
+    private int mFlexWrap;
+
+    /**
+     * The current value of the {@link JustifyContent}, the default value is
+     * {@link JustifyContent#FLEX_START}.
+     */
+    private int mJustifyContent;
+
+    /**
+     * The current value of the {@link AlignItems}, the default value is
+     * {@link AlignItems#STRETCH}.
+     */
+    private int mAlignItems;
+
+    /**
+     * The current value of the {@link AlignContent}, the default value is
+     * {@link AlignContent#STRETCH}.
+     */
+    private int mAlignContent;
+
+    /**
+     * Creates a default FlexboxLayoutManager.
+     *
+     * @param context Current context, will be used to access resources.
+     */
+    public FlexboxLayoutManager(Context context) {
+        this(context, FlexDirection.ROW, FlexWrap.WRAP);
+    }
+
+    public FlexboxLayoutManager(Context context, @FlexDirection int flexDirection) {
+        this(context, flexDirection, FlexWrap.WRAP);
+    }
+
+    /**
+     * @param context Current context, will be used to access resources.
+     */
+    public FlexboxLayoutManager(Context context, @FlexDirection int flexDirection,
+            @FlexWrap int flexWrap) {
+        setFlexDirection(flexDirection);
+        setFlexWrap(flexWrap);
+        setAutoMeasureEnabled(true);
+    }
+
+    /**
+     * Constructor used when layout manager is set in XML by RecyclerView attribute
+     * "layoutManager". No corresponding the attributes for the {@code orientation},
```

Comment:
Might be good to explain how the attributes are mapped. eg: "
<dl><dt>android:orientation HORIZONTAL/VERTICAL</dt><dd>FlexDirection.ROW/COLUMN*</dd><dt>android.support.v7.recyclerview:reverseLayout</dt><dd>FlexDirection.*_REVERSE</dd><dt>android.support.v7.recyclerview:stackFromEnd</dt><dd>FlexWrap.WRAP*</dd>"